design-proposals/feature-lifecycle.md

```diff
@@ -0,0 +1,306 @@
+  discontinuation based on the approved proposal.
+  All details of the plan should be provided to allow users and possibly
+  down-stream projects to adjust.
+  Use all community media options to distribute this information
+  (e.g. mailing list, slack channel, community meetings).
+- Deprecation warnings: Add deprecation warnings at runtime to warn users
+  that the feature is planned to be removed.
+  Warnings should be raised when:
+  - Feature API fields are accessed.
+  - Feature FG is activated.
+  - Behavior related to the feature is detected (optional).
+- Removal: Feature removal involves removing the core functionality
+  of a feature and its exposed API.
+  - The core implementation can be removed in two steps:
+    - The FG is removed by assuring it is never reported as set
+      (i.e. even if it is left by the operator configured, internally
+       it is ignored).
+      At this stage, the core implementation will follow the FG conditions
+      and therefore from the outside the feature is inactive.
+    - In case there are no side effects, the core implementation code can
+      be removed.
+  - The API types are not to be removed, as it may have implications
+    with the underlying storage which has already persisted them.
+    Kubernetes has not removed fields, it just kept them around with the
+    warning that they have been deprecated and no longer available.
+
+    While keeping fields around for a period of a release or two makes
+    sense, beyond a limited period it adds a burden on dragging leftover
+    fields around to eternity.
+
+> **Note**: The only reference seen on why fields should not be removed
+> was mentioned [here](https://github.com/kubernetes/kubernetes/issues/52185).
+> But it is unclear if this is relevant for Alpha stage features.
+
+### Exceptions
+While the project strives to maintain a stable contract with its users,
+there may be scenarios where the policy described here will not be a fit.
+
+Therefore, it should be acceptable to have exceptions from time to time
+given a very good reasoning and an agreement from 2/3 of the project
+maintainers (also known as "approvers").
+
+## Implementation Phases
+- Add a section in the
+  [design proposal template](https://github.com/kubevirt/community/blob/main/design-proposals/proposal-template.md)
+  that describes the planned timelines for the feature stages.
+  Add a reference to the feature-lifecycle documentation to assure contributors
+  know the process and policy.
+- Prepare a user-facing document that describes the usability implications of
+  this feature lifecycle.
+
+## Open Issues
+1. Removal of fields from the CRD: Is it relevant for Alpha
+   and Beta features?
+2. The details on how exactly to mark an Alpha/Beta field.
+   Should we describe it in this proposal or leave it to a follow-up one?
+3. Can Alpha features get introduced in z versions?
+4. CI & E2E restrictions:
+   1. Run e2e tests of Alpha features in their own separate job
```

Comment:
Yes.
These will not need 3 k8s version coverage and their FGs will not get mixed with the other runs.

We currently either enable globally FGs or we enable it per test. When we enable it per test, we are forced to run these serially (because we touch the KV CR).